app/services/expression_evaluators/measure_unit.rb

```diff
@@ -40,24 +40,18 @@ def value
     end
 
     def total_quantity
-      measure_unit_answers.first[:answer].to_f
+      presented_unit[:answer].to_f
     end
 
-    def measure_unit_answers
-      @measure_unit_answers ||= measure_applicable_units.map do |unit, values|
-        {
-          answer: user_session.measure_amount[unit.downcase.to_s],
-          unit: values['unit'],
-        }
-      end
+    def presented_unit
+      @presented_unit ||= {
+        answer: user_session.measure_amount[component.unit.downcase.to_s],
+        unit: applicable_unit['unit'],
+      }
     end
 
-    def measure_applicable_units
-      units = ApplicableMeasureUnitMerger.new.call
-
-      units.select do |_unit, values|
-        values['measure_sids'].include?(measure.id)
-      end
+    def applicable_unit
+      ApplicableMeasureUnitMerger.new.call[component.unit]
```

Comment:
Just pull out the unit defined by the component rather than looking up the unit by the id of either the component or the measure condition component